integration/simple-cache-client.test.ts

```diff
@@ -60,87 +67,151 @@ describe('SimpleCacheClient.ts Integration Tests', () => {
     await withCache(momento, cacheName, async () => {
       await momento.set(cacheName, 'key', 'value');
       const res = await momento.get(cacheName, 'key');
-      expect(res.text()).toEqual('value');
+      expect(res instanceof CacheGet.Error).toEqual(false);
+      if (res instanceof CacheGet.Hit) {
+        expect(res.valueString()).toEqual('value');
+      }
     });
   });
-  it('should throw CacheNotFoundError if deleting a non-existent cache', async () => {
+  it('should return NotFoundError if deleting a non-existent cache', async () => {
     const cacheName = v4();
     const momento = new SimpleCacheClient(AUTH_TOKEN, 1111);
-    await expect(momento.deleteCache(cacheName)).rejects.toThrow(NotFoundError);
+    const deleteResponse = await momento.deleteCache(cacheName);
+    expect(deleteResponse instanceof DeleteCache.Error).toEqual(true);
+    if (deleteResponse instanceof DeleteCache.Error) {
+      expect(deleteResponse.errorCode()).toEqual(
+        MomentoErrorCode.NOT_FOUND_ERROR
+      );
+    }
   });
-  it('should throw CacheAlreadyExistsError if trying to create a cache that already exists', async () => {
+  it('should return AlreadyExists response if trying to create a cache that already exists', async () => {
     const cacheName = v4();
     const momento = new SimpleCacheClient(AUTH_TOKEN, 1111);
     await withCache(momento, cacheName, async () => {
-      await expect(momento.createCache(cacheName)).rejects.toThrow(
-        AlreadyExistsError
-      );
+      const createResponse = await momento.createCache(cacheName);
+      expect(createResponse instanceof CreateCache.AlreadyExists).toEqual(true);
     });
   });
-
   it('should create 1 cache and list the created cache', async () => {
     const cacheName = v4();
     const momento = new SimpleCacheClient(AUTH_TOKEN, 1111);
     await withCache(momento, cacheName, async () => {
-      const caches = (await momento.listCaches()).getCaches();
-      const names = caches.map(c => c.getName());
-      expect(names.includes(cacheName)).toBeTruthy();
+      const listResponse = await momento.listCaches();
+      expect(listResponse instanceof ListCaches.Error).toEqual(false);
+      if (listResponse instanceof ListCaches.Success) {
+        const caches = listResponse.getCaches();
+        const names = caches.map(c => c.getName());
+        expect(names.includes(cacheName)).toBeTruthy();
+      }
```

Comment:
```suggestion
      expect(listResponse instanceof ListCaches.Error).toEqual(false);
      expect(listResponse instanceof ListCaches.Success).toEqual(true)

      const names = listResponse.getCaches().map(c => c.getName());
      expect(names.includes(cacheName)).toBeTruthy();
```

Reply:
`getCaches` is only available if the type of the response is `Success`.  Putting the code inside of the instanceof guard statement allows the compiler to smart-cast to `Success` and thus makes the `getCaches()` method available when it would not otherwise be available.

My expectation would be that the version of the code you pasted here would not compile, because the compiler will only be able to infer the type of `listResponse` to be `ListCachesResponse`, and that class does not have a method called `getCaches`.

Reply:
Yes, the `if` is necessary for the cast.

Reply:
I do agree with @schwern though that the positive assertion is better than the negative ones.  However unlikely, if we were to ever add more subclasses to the response type, then without the `true` assertion for the `Success` type, the test could be a false positive.

Reply:
I was totally assuming these responses would always be binary success or failure, but I take your point about there possibly being other response types in the future though. I'll fix these up.

Reply:
99.9% chance that your assumption is fine :)  but might as well just make the tests 💯